core/src/testils/mod.rs

```diff
@@ -13,16 +13,36 @@ pub mod byte_arrays {
 
 pub use self::packet::*;
 pub use self::rvg::*;
-pub use crate::dpdk::{Mempool, SocketId, MEMPOOL};
 
-use crate::dpdk::eal_init;
+use crate::dpdk::{self, Mempool, SocketId, MEMPOOL};
+use std::ptr;
 use std::sync::Once;
 
 static TEST_INIT: Once = Once::new();
 
-/// Run once initialization of EAL for `cargo test`
+/// Run once initialization of EAL for `cargo test`.
 pub fn cargo_test_init() {
     TEST_INIT.call_once(|| {
-        eal_init(vec!["nb2_test".to_owned()]).unwrap();
+        dpdk::eal_init(vec!["nb2_test".to_owned()]).unwrap();
     });
 }
+
+/// A handle that keeps the mempool in scope for the duration of the test. It
+/// will unset the thread bound mempool on drop.
+#[allow(dead_code)]
```

Comment:
why is this here if it's used within? does the macro not pick it up then?

Reply:
the field `inner` is never used by anything, but just held onto so it lives for the duration of the test.